src/tensors/tensoroperations.jl

```diff
@@ -52,7 +52,9 @@ end
 
 function TO.tensoradd_type(TC, A::AbstractTensorMap, ::Index2Tuple{N₁,N₂},
                            ::Bool) where {N₁,N₂}
-    M = similarstoragetype(A, TC)
+    sst = sectorscalartype(sectortype(A))
+    Tnew = sst <: Real ? TC : VectorInterface.promote_add(TC, scalartype(A), sst)
+    M = similarstoragetype(A, Tnew)
```

Comment:
@lkdvos , if we can completely handle the complex `sectorscalartype` by simply modifying `tensoradd_type` and `tensorcontract_type` that would be my preferred option. That is what these functions are for, create a tensor that can support data of scalar type `TC` that will be obtained from `A` by doing some permutation. I think the fact that the data will be further scaled with complex scalar coefficients is an intrinsic part of our tensor type, and therefore this is where this should be handled. I just find it hard to reason about whether this covers all cases within the `@tensor` macro, but I think it should.

We can even have a separate promotion step of `TC` if we ever want to take more information about `sectorscalartype` into account (e.g. loss of precision etc), but for now I would keep it as simple as possible, e.g.:

```suggestion
    I = sectortype(A)
    M = similarstoragetype(A, sectorscalartype(I) <: Real ? TC : complex(TC))
```